components/selectors/matching.rs

```diff
@@ -466,7 +452,10 @@ fn matches_simple_selector<E>(
         }
         SimpleSelector::Negation(ref negated) => {
             !negated.iter().all(|s| {
-                matches_complex_selector(s, element, parent_bf, relations, flags)
+                match matches_complex_selector_internal(s, element, relations, flags) {
```

Comment:
I think it's not a super-clear win, and I wanted to prevent the filter from being misused, but I guess I can do that too.